autoload/ale/events.vim

```diff
@@ -11,8 +11,32 @@ function! ale#events#QuitRecently(buffer) abort
     return l:time && ale#util#ClockMilliseconds() - l:time < 1000
 endfunction
 
+function! ale#events#EnterEvent(buffer) abort
+    " When entering a buffer, we are no longer quitting it.
+    call setbufvar(a:buffer, 'ale_quitting', 0)
+
+    " Save the original filetype for later comparison.
+    let l:filetype = getbufvar(a:buffer, '&filetype')
+    call setbufvar(a:buffer, 'ale_original_filetype', l:filetype)
+
+    " Don't run costly, pointless operations on startup.
+    if v:vim_did_enter
+        " Update highlights.
+        call ale#highlight#UpdateHighlights()
+
+        " Apply fix if one was run in the background.
+        call ale#fix#ApplyQueuedFixes(a:buffer)
```

Comment:
Calling this function will cause `autoload/ale/fix.vim` to be loaded, even if you don't use ALE for fixing things. You could wrap this call in an `exists` check like we do for `ale#statusline#Update` in `autoload/ale/engine.vim`.

Reply:
After reading code from different files, I think you should probably move this call back to where it was in the `fix.vim` file. We will still need to run this function if `g:ale_enabled` is `0`, and if you run `:ALEFix` and switch buffers quickly enough.